libs/ts-rest/core/src/lib/paths.spec.ts

```diff
@@ -92,4 +92,38 @@ describe('insertParamsIntoPath', () => {
 
     expect(result).toBe('/1');
   });
+
+  it('should insert optional params into path with many params', () => {
+    const path = '/post/:id?/comments/:commentId?/:commentId2?';
+
+    const params = {
+      commentId: '2',
+      commentId2: '3',
+      id: '1',
+    };
+
+    const result = insertParamsIntoPath({ path, params });
+
+    expect(result).toBe('/post/1/comments/2/3');
+  });
+
+  it('should insert optional params into path with no params', () => {
+    const path = '/post/:id?/comments/:commentId?/:commentId2?';
+
+    const result = insertParamsIntoPath({ path, params: {} });
+
+    expect(result).toBe('/post/comments/');
```

Comment:
Not sure how to handle the trailing slash here, both `:commentId` and `:commentId2` are missing.

The function replaces two consecutive slashes with a single one but it doesn't handle this case where more than 1 missing optional params appear in the end of the path.